### ISSUE TITLE
Can't generate for Xeon E5-2660 ES

### ISSUE THREAD

**Issue by mirkoevo** [Piker-Alpha/ssdtPRGen.sh#238]
Hi,

I'm trying to get XCPM support with a Xeon E5-2660ES and a Asus R4G but I can't generate the SSDT:

olive:ssdtPRGen.sh mirko$ uname -a
Darwin olive.local 14.5.0 Darwin Kernel Version 14.5.0: Wed Jul 29 02:26:53 PDT 2015; root:xnu-2782.40.9~1/RELEASE_X86_64 x86_64

gUserDefinedCPUList=(
'E5-2660ES v2’,95,1200,2100,2900,10,20  # original
'E5-2660ESt v2’,95,1200,2500,3000,10,20  # overclocked
)

olive:ssdtPRGen.sh mirko$ sh ssdtPRGen.sh -p "E5-2660ESt v2" -x 1 -w 3 -c 1

ssdtPRGen.sh v0.9  Copyright (c) 2011-2012 by † RevoGirl
             v6.6  Copyright (c) 2013 by † Jeroen
##              v18.9 Copyright (c) 2013-2016 by Pike R. Alpha

Bugs > https://github.com/Piker-Alpha/ssdtPRGen.sh/issues <

System information: Mac OS X 10.10.5 (14F27)
Brandstring: "Genuine Intel(R) CPU @ 2.10GHz"

Override value: (-p) processor model, now using: E5-2660ESt v2!
Override value: (-x) XCPM mode, now set to: 1!
Override value: (-w) Ivy Bridge workarounds, now set to: 3!
sed: 1: "s/^[\n]*
Error://
": unterminated substitute pattern

Error:
Aborting ...
Done.

That's a bit old, but it is the same:

olive:ssdtPRGen.sh mirko$ cat /var/log/system.log | grep -e 'ACPI' -e 'X86PlatformPlugin' -e 'X86PlatformShim' -e 'CPUMaxNonTurboPState' -e 'Max efficient' -e 'XCPM' -e 'IOPPF' -e 'AGPM' -e 'IOPlatformPluginLegacy'

Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=0 LocalApicId=0 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=2 LocalApicId=2 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=4 LocalApicId=4 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=6 LocalApicId=6 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=8 LocalApicId=8 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=10 LocalApicId=16 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=12 LocalApicId=18 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=14 LocalApicId=20 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=16 LocalApicId=22 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=18 LocalApicId=24 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=1 LocalApicId=1 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=3 LocalApicId=3 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=5 LocalApicId=5 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=7 LocalApicId=7 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=9 LocalApicId=9 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=11 LocalApicId=17 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=13 LocalApicId=19 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=15 LocalApicId=21 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=17 LocalApicId=23 Enabled
Oct 25 07:12:33 localhost kernel[0]: AppleACPICPU: ProcessorId=19 LocalApicId=25 Enabled
Oct 25 07:12:33 localhost kernel[0]: ACPI: sleep states S3 S4 S5
Oct 25 07:12:33 localhost kernel[0]: Got boot device = IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/SAT0@1F,2/AppleAHCI/CHN0@0/IOAHCIDevice@0/AppleAHCIDiskDriver/IOAHCIBlockStorageDevice/IOBlockStorageDriver/Samsung SSD 840 EVO 500GB Media/IOGUIDPartitionScheme/Macintosh SSD@2
Oct 25 07:12:33 olive kernel[0]: IOPPF: AppleIntelCPUPowerManagement mode
Oct 25 07:12:33 olive kernel[0]: X86PlatformShim::sendPStates - Success!
Oct 25 07:12:33 olive kernel[0]: X86PlatformShim::sendStepper - Stepper dictionary from plist is NULL!
Oct 25 07:12:33 olive kernel[0]: X86PlatformShim::start - Failed to send stepper
Oct 25 07:17:21 olive.local sudo[625]:    mirko : TTY=ttys015 ; PWD=/Volumes/Leopard/Users/mirko/Downloads/R4G ; USER=root ; COMMAND=/bin/rm /EFI/CLOVER/ACPI/patched/DSDT.aml
Oct 25 08:05:18 olive.local com.apple.kextcache[1109]: kext file:///System/Library/Extensions/AppleACPIPS2Nub.kext/ is in hash exception list, allowing to load
Oct 25 08:06:23 olive.local com.apple.kextcache[1122]: kext file:///System/Library/Extensions/AppleACPIPS2Nub.kext/ is in hash exception list, allowing to load

AppleIntelInfo.kext v1.2 Copyright © 2012-2015 Pike R. Alpha. All rights reserved
## Settings:

logMSRs............................: 1
logIGPU............................: 0
logIntelRegs.......................: 1
logCStates.........................: 1
logIPGStyle........................: 1
InitialTSC.........................: 0x88a086503ab
MWAIT C-States.....................: 4384
## Model Specific Regiters

MSR_CORE_THREAD_COUNT......(0x35)  : 0xA0014
MSR_PLATFORM_INFO..........(0xCE)  : 0xC106C811500
MSR_PMG_CST_CONFIG_CONTROL.(0xE2)  : 0x401
MSR_PMG_IO_CAPTURE_BASE....(0xE4)  : 0x10414
IA32_MPERF.................(0xE7)  : 0x512716081E
IA32_APERF.................(0xE8)  : 0x788820AC2C
MSR_FLEX_RATIO.............(0x194) : 0x0
MSR_IA32_PERF_STATUS.......(0x198) : 0x1D4D00001900
MSR_IA32_PERF_CONTROL......(0x199) : 0x1900
IA32_CLOCK_MODULATION......(0x19A) : 0x0
IA32_THERM_STATUS..........(0x19C) : 0x88420000
IA32_MISC_ENABLES..........(0x1A0) : 0x850089
MSR_MISC_PWR_MGMT..........(0x1AA) : 0x400001
MSR_TURBO_RATIO_LIMIT......(0x1AD) : 0x191919191A1B1C1D
IA32_ENERGY_PERF_BIAS......(0x1B0) : 0x0
MSR_POWER_CTL..............(0x1FC) : 0x2104005B
MSR_RAPL_POWER_UNIT........(0x606) : 0xA1003
MSR_PKG_POWER_LIMIT........(0x610) : 0x68390005A82F8
MSR_PKG_ENERGY_STATUS......(0x611) : 0x1C9D720B
MSR_PKG_POWER_INFO.........(0x614) : 0x2F06E001B002F8
MSR_PP0_CURRENT_CONFIG.....(0x601) : 0x14149480001FFF
MSR_PP0_POWER_LIMIT........(0x638) : 0x80000000
MSR_PP0_ENERGY_STATUS......(0x639) : 0x78741401
MSR_PP0_POLICY.............(0x63a) : 0x0
MSR_PKGC3_IRTL.............(0x60a) : 0x0
MSR_PKGC6_IRTL.............(0x60b) : 0x0
MSR_PKG_C2_RESIDENCY.......(0x60d) : 0x2A654860994
MSR_PKG_C6_RESIDENCY.......(0x3f9) : 0x47062FFE93
IA32_TSC_DEADLINE..........(0x6E0) : 0x88A0A134F01
PCH device.................: 0x1D418086
## CPU Ratio Info:

CPU Low Frequency Mode.............: 1200 MHz
CPU Maximum non-Turbo Frequency....: 2100 MHz
CPU Maximum Turbo Frequency........: 2900 MHz
CPU P-States [ 23 27 (29) ]
CPU C6-Cores [ 0 1 5 7 9 12 13 14 15 16 ]
CPU P-States [ (12) 21 23 27 29 ]
CPU C6-Cores [ 0 1 5 7 9 10 11 12 13 14 15 16 ]
CPU P-States [ (12) 21 23 25 27 29 ]
CPU C6-Cores [ 0 1 2 5 7 8 9 10 11 12 13 14 15 16 ]
CPU P-States [ 12 21 22 23 25 27 (29) ]
CPU C6-Cores [ 0 1 2 4 5 7 8 9 10 11 12 13 14 15 16 ]
CPU P-States [ 12 21 22 23 25 27 (28) 29 ]
CPU C6-Cores [ 0 1 2 4 5 6 7 8 9 10 11 12 13 14 15 16 ]
CPU C6-Cores [ 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 ]

**Comment by Piker-Alpha** [comment]
Please attach a zipped up copy of the ~/Library/ssdtPRGen folder.

**Comment by mirkoevo** [comment]
[ssdtPRGen.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/358001/ssdtPRGen.zip)

**Comment by Piker-Alpha** [comment]
What SMBIOS (board-id) model are you using?

Then this output in your log file:

IOPPF: AppleIntelCPUPowerManagement mode
X86PlatformShim::sendPStates - Success!

Which is fine. Then this:

X86PlatformShim::sendStepper - Stepper dictionary from plist is NULL!
X86PlatformShim::start - Failed to send stepper

This error is caused by missing _StepContextDict_ data that should be in this plist: IOPlatformPluginFamily.kext/C_/P_/X86PlatformPlugin.kext/C_/R_/Mac-[board-id].plist

In short. Using a model/board-id without _StepContextDict_ data in the target plist is an error.

Tip: Your processor doesn't support XCPM and thus adding FrequencyVectors data is not required. Won't help a bit. Just in case you've done that.

By the way. Are you using Clovers FakeCPUID feature? If yes, what model?

**Comment by mirkoevo** [comment]
I am not using the fake id of Clover and I think I am using MacPro 6,1 but can't confirm until the end of the month since I am on vacation. Do you know if I can enable sleep/wake feature successfully in some way with this processor due to its limitations? Thank you   

-- Best regards,Mirko  

On 20 luglio 2016 13:39:51 CEST, Pike R. Alpha notifications@github.com wrote:What SMBIOS (board-id) model are you using?  Then this output in your log file:  IOPPF: AppleIntelCPUPowerManagement mode X86PlatformShim::sendPStates - Success!  Which is fine. Then this:  X86PlatformShim::sendStepper - Stepper dictionary from plist is NULL! X86PlatformShim::start - Failed to send stepper  This error is caused by missing StepContextDict data that should be in this plist: IOPlatformPluginFamily.kext/C/P/X86PlatformPlugin.kext/C/R/Mac-[board-id].plist  In short. Using a model/board-id without StepContextDict data in the target plist is an error.  Tip: Your processor doesn't support XCPM and thus adding FrequencyVectors data is not required. Won't help a bit. Just in case you've done that.  By the way. Are you using Clovers FakeCPUID feature? If yes, what model?  —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by thearmlesswonder** [comment]
Hello, I am also having trouble.  I have a single CPU E5-1620 ( 4 cores of course ).  I can see its on your list for sandy bridge .  Basically I would like to get my CPU states working, its currently flat line at 3.6 and never changes.  Power a tiny bit and temp.  Temp is at 140F or 63C, not sure that is good all the time.  Ive been at it for hours now and have read a bunch of your post.  To keep this short.  I have this problem:
No ACPI Processor declarations found in the DSDT!
https://github.com/Piker-Alpha/ssdtPRGen.sh/issues/17

I have trird stuff from here too:
https://github.com/Piker-Alpha/ssdtPRGen.sh/issues/80

I went crazy with setting flags but no luck.  

Thoughts ? Take your time.  Thank you for your efforts.

**Comment by thearmlesswonder** [comment]
Today I noticed its jumping from 3.60 to 3.36, up and down.  I managed to get a SSDT to work from your script some how but I don't think its right.  I followed some other guide.  Still would like to make this run a lot better.

**Comment by Piker-Alpha** [comment]
Hi there,

I would like to fix the "_No ACPI Processor declarations found in the DSDT!_" to start with, so that I can reproduce the SSDT that you are generating. Please attach your DSDT.aml and APIC.aml (see ˜/Library/ACPI/ folder).

Thanks!

**Comment by thearmlesswonder** [comment]
Hey Piker-Alpha,  Thanks for taken the time.  Ok I am using Clover r3726 and I just Pressed F3 or F4 to get Clover to generate the ACPI - orgin.  Then I copied every thing from there to ˜/Library/ACPI/ folder.  Not sure if thats right ? It was blank and I thought it said it was missing when I was running the script.  Let me know it thats right or not.

I attached them below.

[origin.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/475790/origin.zip)

Note: when I woke up from sleep today, I watched the intel power Gadget and saw my Watts, and Temp were a lot lower.  Also Feq. changes evey sec. to 3.12, 3.36, back to 3.60.  and I just saw 2.88.  Weird, because it moves vary few or its flat lined at 3.60 upon boot up and I am running nothing or something.  Temp and watts are vary high on boot.

**Comment by Piker-Alpha** [comment]
The script failed to locate the required processor declarations in your DSDT. The reason for this is that they are not there. The can however be found in the SSDT-1. But we did not check this file.

The solution is – so I hope – is to check the SSDT's for SCK0 and when one or more files are matching, then we also convert them to a postscript format like we do with the DSDT.

Please let me know if this works for you.
